testsuite/MDAnalysisTests/coordinates/test_reader_api.py

```diff
@@ -133,96 +138,91 @@ class _Multi(_TestReader):
 
 
 class TestMultiFrameReader(_Multi):
+
     __test__ = True
-    def _check_slice(self, start, stop, step):
+
+
+    @pytest.mark.parametrize('start, stop, step', [
+        (None, None, None),  # blank slice
+        (None, 5, None),  # set end point
+        (2, None, None),  # set start point
+        (2, 5, None),  # start & end
+        (None, None, 2),  # set skip
+        (None, None, -1),  # backwards skip
+        (0, 10, 1),
+        (0, 10, 2),
+        (None, 20, None),  # end beyond real end
+        (None, 20, 2),  # with skip
+        (0, 5, 2),
+        (5, None, -1),
+        (None, 5, -1),
+        (100, 10, 1),
+        (-10, None, 1),
+        (100, None, -1),  # beyond real end
+        (100, 5, -20),
+        (5, 1, 1),  # Stop less than start
+        (1, 5, -1),  # Stop less than start
+        (-100, None, None),
+        (100, None, None),  # Outside of range of trajectory
+        (-2, 10, -2)
+    ])
+    def test_slice(self, start, stop, step, reader):
         """Compare the slice applied to trajectory, to slice of list"""
-        res = [ts.frame for ts in self.reader[start:stop:step]]
+        res = [ts.frame for ts in reader[start:stop:step]]
         ref = self.reference[start:stop:step]
 
         assert_equal(res, ref)
 
-    def test_slices(self):
-        for start, stop, step in [
-            (None, None, None),  # blank slice
-            (None, 5, None),  # set end point
-            (2, None, None),  # set start point
-            (2, 5, None),  # start & end
-            (None, None, 2),  # set skip
-            (None, None, -1),  # backwards skip
-            (0, 10, 1),
-            (0, 10, 2),
-            (None, 20, None),  # end beyond real end
-            (None, 20, 2),  # with skip
-            (0, 5, 2),
-            (5, None, -1),
-            (None, 5, -1),
-            (100, 10, 1),
-            (-10, None, 1),
-            (100, None, -1),  # beyond real end
-            (100, 5, -20),
-            (5, 1, 1), # Stop less than start
-            (1, 5, -1), # Stop less than start
-            (-100, None, None),
-            (100, None, None), # Outside of range of trajectory
-            (-2, 10, -2)
-        ]:
-            yield self._check_slice, start, stop, step
-
-    def test_slice_VE_1(self):
+    def test_slice_VE_1(self, reader):
         def sl():
-            return list(self.reader[::0])
+            return list(reader[::0])
 
         assert_raises(ValueError, sl)
 
-    def test_slice_TE_1(self):
+    def test_slice_TE_1(self, reader):
         def sl():
-            return list(self.reader[1.2:2.5:0.1])
+            return list(reader[1.2:2.5:0.1])
 
         assert_raises(TypeError, sl)
 
-    def _check_getitem(self, sl):
-        res = [ts.frame for ts in self.reader[sl]]
+    @pytest.mark.parametrize('sl', [
```

Comment:
I think this has to do with static checking, If I define the fixture in the subclass itself, then I'm able to access it in the parameterize call.